trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/validate-payment-details-filters.test.ts

```diff
@@ -0,0 +1,240 @@
+        expect(result.paymentCurrencyErrorMessage).toEqual(expectedPaymentCurrencyErrorMessage);
+        expect(result.paymentReferenceErrorMessage).toBeUndefined();
+      });
+    });
+
+    describe('when payment reference is invalid', () => {
+      it('should return an object with paymentReferenceErrorMessage and errorSummary', () => {
+        // Arrange
+        const filters = {
+          facilityId: '1234',
+          paymentCurrency: CURRENCY.GBP,
+          paymentReference: 'ABC',
+        };
+        const originalUrl = 'http://example.com?paymentDetailsFacilityId=1234&paymentDetailsPaymentCurrency=GBP&paymentDetailsPaymentReference=ABC';
+
+        const expectedPaymentReferenceErrorMessage = 'Payment reference must be blank or contain a minimum of 4 characters';
+
+        // Act
+        const result = validatePaymentDetailsFilters(originalUrl, filters);
+
+        // Assert
+        expect(result.errorSummary).toEqual([
+          {
+            text: expectedPaymentReferenceErrorMessage,
+            href: '#payment-details-payment-reference-filter',
+          },
+        ]);
+        expect(result.facilityIdErrorMessage).toBeUndefined();
+        expect(result.paymentCurrencyErrorMessage).toBeUndefined();
+        expect(result.paymentReferenceErrorMessage).toEqual(expectedPaymentReferenceErrorMessage);
+      });
+    });
+
+    describe('when all filters are invalid', () => {
+      it('should return an object with all error messages and errorSummary', () => {
+        // Arrange
+        const filters = {
+          facilityId: 'INVALID',
+          paymentCurrency: 'INVALID',
+          paymentReference: 'ABC',
+        };
+        const originalUrl = 'http://example.com?paymentDetailsFacilityId=INVALID&paymentDetailsPaymentCurrency=INVALID&paymentDetailsPaymentReference=ABC';
+
+        // Act
+        const result = validatePaymentDetailsFilters(originalUrl, filters);
+
+        // Assert
+        expect(result.errorSummary).toHaveLength(3);
+        expect(result.facilityIdErrorMessage).toBeDefined();
+        expect(result.paymentCurrencyErrorMessage).toBeDefined();
+        expect(result.paymentReferenceErrorMessage).toBeDefined();
+      });
+    });
+
+    describe('when original URL does not include filter parameters', () => {
+      it('should return an object with all error messages and errorSummary', () => {
+        // Arrange
+        const filters = {
+          facilityId: '1234',
+          paymentCurrency: CURRENCY.GBP,
+          paymentReference: 'REF123',
+        };
+        const originalUrl = 'http://example.com';
+
+        // Act
+        const result = validatePaymentDetailsFilters(originalUrl, filters);
+
+        // Assert
+        expect(result.errorSummary).toHaveLength(3);
+        expect(result.facilityIdErrorMessage).toBeDefined();
+        expect(result.paymentCurrencyErrorMessage).toBeDefined();
+        expect(result.paymentReferenceErrorMessage).toBeDefined();
```

Comment:
Can we test the whole `result` and use `toEqual` assertions instead of `toBeDefined` please?